service/awsconfig/v5/resource/cloudformation/update.go

```diff
@@ -50,6 +51,25 @@ func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
 	return patch, nil
 }
 
+func (r *Resource) computeUpdateState(customObject v1alpha1.AWSConfig, desiredState StackState) (cloudformation.UpdateStackInput, error) {
+	mainTemplate, err := r.getMainGuestTemplateBody(customObject)
+	if err != nil {
+		return cloudformation.UpdateStackInput{}, microerror.Mask(err)
+	}
+
+	updateState := cloudformation.UpdateStackInput{
+		Capabilities: []*string{
+			// CAPABILITY_NAMED_IAM is required for updating IAM roles (worker
+			// policy).
+			aws.String(namedIAMCapability),
+		},
+		StackName:    aws.String(desiredState.Name),
+		TemplateBody: aws.String(mainTemplate),
+	}
+
+	return updateState, nil
+}
```

Comment:
There is now a reusable method for computing the update state which we use for scaling and updates respectively.